tests/test_phirgen.py

```diff
@@ -201,3 +263,171 @@ def test_conditional_measure() -> None:
+    # conditionals
+    c.add_c_not_to_registers(c1, c2, condition=Bit("c0", 0))
+    c.add_c_not_to_registers(c1, c1, condition=Bit("c0", 0))
+    phir = json.loads(pytket_to_phir(c))
+    assert phir["ops"][3] == {
+        "//": "AND (*3) c0[0], c1[0], c2[0], c0[1], c1[1], c2[1], c0[2], c1[2], c2[2];"
+    }
+    assert phir["ops"][4] == {
+        "cop": "=",
+        "returns": ["c2"],
+        "args": [{"cop": "&", "args": ["c0", "c1"]}],
+    }
+    assert phir["ops"][5] == {
+        "//": "NOT (*4) c1[0], c2[0], c1[1], c2[1], c1[2], c2[2], c1[3], c2[3];"
+    }
+    assert phir["ops"][6] == {
+        "cop": "=",
+        "returns": ["c2"],
+        "args": [{"cop": "~", "args": ["c1"]}],
+    }
+    assert phir["ops"][7] == {
+        "//": "OR (*3) c0[0], c1[0], c2[0], c0[1], c1[1], c2[1], c0[2], c1[2], c2[2];"
+    }
+    assert phir["ops"][8] == {
+        "cop": "=",
+        "returns": ["c2"],
+        "args": [{"cop": "|", "args": ["c0", "c1"]}],
+    }
+    assert phir["ops"][9] == {
+        "//": "XOR (*4) c1[0], c2[0], c1[1], c2[1], c1[2], c2[2], c1[3], c2[3];"
+    }
+    assert phir["ops"][10] == {
+        "cop": "=",
+        "returns": ["c2"],
+        "args": [{"cop": "^", "args": ["c1", "c2"]}],
+    }
+    assert phir["ops"][12] == {
+        "block": "if",
+        "condition": {"cop": "==", "args": [["c0", 0], 1]},
+        "true_branch": [
+            {"cop": "=", "returns": ["c2"], "args": [{"cop": "~", "args": ["c1"]}]}
+        ],
+    }
+    assert phir["ops"][14] == {
+        "block": "if",
+        "condition": {"cop": "==", "args": [["c0", 0], 1]},
+        "true_branch": [
+            {"cop": "=", "returns": ["c1"], "args": [{"cop": "~", "args": ["c1"]}]}
+        ],
+    }
+
+
+def test_irregular_multibit_ops() -> None:
+    """From https://github.com/CQCL/pytket-phir/pull/162#discussion_r1555807863 ."""
+    c = Circuit()
+    areg = c.add_c_register("a", 2)
+    breg = c.add_c_register("b", 2)
+    creg = c.add_c_register("c", 2)
+    c.add_c_and_to_registers(areg, breg, creg)
+    mbop = c.get_commands()[0].op
+    c.add_gate(mbop, [areg[0], areg[1], breg[0], breg[1], creg[0], creg[1]])
+
+    phir = json.loads(pytket_to_phir(c))
+    assert phir["ops"][3] == {"//": "AND (*2) a[0], b[0], c[0], a[1], b[1], c[1];"}
+    assert phir["ops"][4] == {
+        "cop": "=",
+        "returns": ["c"],
+        "args": [{"cop": "&", "args": ["a", "b"]}],
+    }
+    assert phir["ops"][5] == {"//": "AND (*2) a[0], a[1], b[0], b[1], c[0], c[1];"}
+    assert phir["ops"][6] == {
+        "block": "sequence",
+        "ops": [
+            {
+                "cop": "=",
+                "returns": [["b", 0]],
+                "args": [{"cop": "&", "args": [["a", 0], ["a", 1]]}],
+            },
+            {
+                "cop": "=",
+                "returns": [["c", 1]],
+                "args": [{"cop": "&", "args": [["b", 1], ["c", 0]]}],
+            },
+        ],
+    }
```

Comment:
Yeah, you can use a sequence block like that. It is just a way to group things. Actually maybe all things that are a sequence of operations should be using this instead of just [] but oh well.

Is this supposed to be something like `b[0],c[1] = [a[0], b[1] ] & [a[1], c[0]]`? If so that looks fine.

If this is all too contrived I wonder if we should allow `list of bits` in addition to creg or bit and arguments and returns... might simplify things...

Reply:
Thanks.

Since the spec currently doesn't support `list of bits`, I'd prefer to stick to this solution since it doesn't require changes in either the phir spec or PECOS.

PS: yeah, it's Lis(t)P(rocessor) but I like to see parens there then not brackets. :)

Reply:
Sounds good. Then if `[b[0],c[1]] = [a[0], b[1] ] & [a[1], c[0]]` was the intent then that looks correct to me.